compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -2497,6 +2524,11 @@ void t_py_generator::generate_serialize_field(ostream& out, t_field* tfield, str
  * @param prefix  String prefix to attach to all fields
  */
 void t_py_generator::generate_serialize_struct(ostream& out, t_struct* tstruct, string prefix) {
+  if(gen_type_hints_) {
+    indent(out) << "if not type(" << prefix << ") is " << type_to_py_type(tstruct)
```

Comment:
I think I agree with @jeking3  that this goes beyond type hinting and is a behavioral change that might deserve its own PR and discussion. Any opposition to backing this part of the diff out so the non-controversial type hints can land?